Doc/c-api/init.rst

```diff
@@ -1279,6 +1279,21 @@ with sub-interpreters:
       Hangs the current thread, rather than terminating it, if called while the
       interpreter is finalizing.
 
+
+.. c:function:: int PyGILState_EnsureOrFail(PyGILState_STATE *state)
+
+   Similar to :c:func:`PyGILState_Ensure`, except that it returns with a status
+   code even in the case of failure. Specifically, it returns ``0`` when the
+   operation succeeded, and ``-1`` otherwise. In contrast to
+   :c:func:`PyGILState_Ensure`, *state* is an argument.
+
+   In the case of failure, it is *unsafe* to use the Python API following the
+   call. Releasing the obtained *state* via :c:func:`PyGILState_Release` should
+   only be done in the case of success.
+
```

Comment:
>> Could you add the mention that these are “not compatible with sub-interpreters” directly to the docs?
> Which API should be recommended instead, if this one is not compatible with sub-interpreters?

I meant repeating the note above `PyGILState_Ensure` that says *The following functions use thread-local storage, and are not compatible with sub-interpreters:*. Users that click the What's New link go directly to the function docs and are quite likely to miss the warning.
Something like:

```suggestion

   Like :c:func:`!PyGILState_Ensure`, this function is not safe to use with
   supinterpreters, since it is not possible to specify which interpreter
   is activated.

```

Reply:
https://github.com/python/cpython/pull/130012 PR should be subinterpreter-friendly, I closed this PR instead.